File: backend/app/routers/workspaces.py
from fastapi import APIRouter

from app.schemas.workspace import Workspace, WorkspaceCreate, WorkspaceRename
from app.services import workspace_service

router = APIRouter(prefix="/workspaces", tags=["workspaces"])


@router.get("", response_model=list[Workspace])
def list_workspaces() -> list[Workspace]:
    return workspace_service.list_workspaces()


@router.post("", response_model=Workspace, status_code=201)
def create_workspace(data: WorkspaceCreate) -> Workspace:
    return workspace_service.create_workspace(data)


@router.patch("/{workspace_id}", response_model=Workspace)
def rename_workspace(workspace_id: str, data: WorkspaceRename) -> Workspace:
    return workspace_service.rename_workspace(workspace_id, data.name)


@router.delete("/{workspace_id}", status_code=204)
def delete_workspace(workspace_id: str) -> None:
    workspace_service.delete_workspace(workspace_id)

File: backend/src/routers/workspaces.ts
import { Router } from 'express';
import {
  listWorkspaces,
  createWorkspace,
  renameWorkspace,
  deleteWorkspace
} from '../services/workspaceService.js';

const router = Router();

router.get('/workspaces', (req, res, next) => {
  try {
    const workspaces = listWorkspaces();
    res.json(workspaces);
  } catch (err) {
    next(err);
  }
});

router.post('/workspaces', (req, res, next) => {
  try {
    const workspace = createWorkspace(req.body);
    res.status(201).json(workspace);
  } catch (err) {
    next(err);
  }
});

router.patch('/workspaces/:workspace_id', (req, res, next) => {
  try {
    const workspace = renameWorkspace(req.params.workspace_id, req.body.name);
    res.json(workspace);
  } catch (err) {
    next(err);
  }
});

router.delete('/workspaces/:workspace_id', (req, res, next) => {
  try {
    deleteWorkspace(req.params.workspace_id);
    res.status(204).end();
  } catch (err) {
    next(err);
  }
});

export default router;
